test/read_mixin_test.rb

```diff
@@ -32,6 +34,47 @@
       assert_equal @post_records, Post.find_by_sql('SELECT * FROM `posts` WHERE (id = 1)')
     end
 
+    describe "it can retrieve one or multiple records through the high-level AR model API (Kasket query[:key] is an Array)" do
+      before do
+        # Actually go to the DB.
+        Comment.unstub(:find_by_sql_without_kasket)
+
+        Kasket.cache.write("#{Comment.kasket_key_prefix}id=1", @comment_database_result[0])
+        Kasket.cache.write("#{Comment.kasket_key_prefix}id=2", @comment_database_result[1])
+
+        Kasket.cache.write("#{Comment.kasket_key_prefix}post_id=1", [
+          "#{Comment.kasket_key_prefix}id=1",
+          "#{Comment.kasket_key_prefix}id=2"
+        ])
+      end
+
+      specify "one record" do
+        Comment.expects(:find_by_sql_without_kasket).never
+
+        out = Comment.find(1)
+        assert_equal @comment_records[0], out
+
+        out = Comment.find_by(id: 1)
+        assert_equal @comment_records[0], out
+
+        out = Comment.where(id: 2).first
+        assert_equal @comment_records[1], out
+      end
+
+      specify "multiple records" do
+        Comment.expects(:find_by_sql_without_kasket).never
+
+        out = Comment.find(1, 2)
+        assert_equal @comment_records, out
+
+        out = Comment.where(id: [1, 2]).to_a
+        assert_equal @comment_records, out
+
+        out = Comment.where(post_id: 1).to_a
+        assert_equal @comment_records, out
+      end
+    end
```

Comment:
This is extra test coverage that I felt I needed to add in order to understand the current behaviour, before the fix.